### PR TITLE
Solution - Text to Morse code

### PR DESCRIPTION
**Pull request by AhmedKhalifa7**
---
name: Solution Review 
about: Code review checklist
---

## Behavior

The function converts a given text into Morse code. It processes each character of the input text and and concerts it to its corresponding Morse code representation.

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [x] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible
- [ ]

### CODE REVIEW

**Review by MohammadRAlSalloum — APPROVED** [review]
Great idea, clean code, following the checklist!

It can not get better!

**Comment by dadishimwe** [comment]
Very creative. I like it!!

**Comment by AhmedKhalifa7** [comment]
I had on test_ with with a wrong value and now I edited it, can you please review it again? And make sure to check the checklist

**Review by kerthnorth — COMMENTED** [review]
Great Job! The test documentation align with tests. The tests also consider numerous scenarios. 

The code solves the problem and raises error if a value does not have corresponding case. Keep up the great work!

File: solutions/text_to_morse.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    >>> text_to_morse("forza milan")
+    '..-. --- .-. --.. .- / -- .. .-.. .- -.'
+
+    >>> text_to_morse("CQD DE MGY")
+    '-.-. --.- -.. / -.. . / -- --. -.--'
+
+    >>> text_to_morse("/-)")
+    '-..-. -....- -.--.-'
+
```
Comment (RaedAlyan) [inline solutions/text_to_morse.py:42]: remove this blank line pls

File: solutions/tests/test_text_to_morse.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+
```
Comment (RaedAlyan) [inline solutions/tests/test_text_to_morse.py:4]: remove this blank line

File: solutions/tests/test_text_to_morse.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (AhmedKhalifa7) [reply solutions/tests/test_text_to_morse.py:1]: > are you working on Linux or windows OS?

No, honestly I really don't understand what it's, I'm just adding it because Evan said so.😂

File: solutions/tests/test_text_to_morse.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  - Defensive tests: assertions, unknown character
+
+Created on 5 1 2025
+
+@author: Ahmed Hussein
+"""
+
+import unittest
+
+
```
[the remark this answers is not in the document]
Comment (AhmedKhalifa7) [reply solutions/tests/test_text_to_morse.py]: All blank lines has been removed, thanks Raed!

**Review by RamaBio20 — COMMENTED** [review]
awesome project idea! ❤️

File: solutions/tests/test_text_to_morse.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (RaedAlyan) [reply solutions/tests/test_text_to_morse.py:1]: please remove it
